### PR TITLE
PPU LLVM upgrade (WIP)

### PR DESCRIPTION
**Pull request by Nekotekina**
First commit only contains some cleanup atm.

### CODE REVIEW

**Comment by alexloxldlol** [comment]
F {PPU[0x1000000] Thread (main_thread)} SIG: Thread terminated due to fatal error: Verification failed (e=0x57): 
(in file d:\a\1\s\Utilities\VirtualMemory.cpp:86)

**Comment by Nekotekina** [comment]
@alexloxldlol Should be fixed

**Comment by alexloxldlol** [comment]
> @alexloxldlol debe repararse

yeah works now

**Comment by Nekotekina** [comment]
Added experimental code

**Comment by Nekotekina** [comment]
At the cost of longer compilation times and higher memory and disk usage, may improve performance in games which previously required interpreter fallback. Sadly I don't remember which.

**Comment by digitaldude555** [comment]
#5831 prison break now works with PPU LLVM.

**Comment by JohnHolmesII** [comment]
Installing the firmware spams "Error: Branch with absolute address" tens of thousands of times.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/4663752/RPCS3.log.gz)

**Comment by Nekotekina** [comment]
Ignore the spam, I'll do something with it later.

**Comment by JohnHolmesII** [comment]
Can confirm @RainbowCookie32's issue from Discord, Demon's Souls crashes the emulator on boot.
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/4663876/RPCS3.log)

**Comment by Dokman** [comment]
i'll try with infamous :)

**Comment by TigiHof** [comment]
L.A. Noire is one of the games that require PPU interpreter fallback. Unfortunately it also crashes the emulator on boot.
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/4664269/RPCS3.log)

**Comment by weter11** [comment]
In a MCLA (BLUS30442) error when compiling
`·E 2:05:59.650236 {PPUW.1.6} PPU: LLVM: [0x00ac7894] Error: Unsupported instruction LSWX.`
Never saw such error before. And a lot more modules now compiled (from 99 to 161). Interesting error, unless someone has CELL CPU. But game is working, same performance.

**Comment by Kravickas** [comment]
NHL® Legacy [BLUS31540] - 5820k haswell no-tsx

·E 0:07:04.570933 {PPUW.1.5} PPU: LLVM: [0x01dc14b0] Error: Unsupported instruction STSWX. Please report.
·E 0:07:04.577462 {PPUW.1.5} PPU: LLVM: [0x01dc20dc] Error: Unsupported instruction LSWX. Please report.

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/4671846/RPCS3.log)

After all PPU is compiled RPCS3 close itself without error.

**Comment by Nekotekina** [comment]
Updated PR, hopefully fixed the crash.

**Comment by Kravickas** [comment]
Crash and error fixed

**Comment by chane2k1** [comment]
Daytona USA started working correctly with LLVM in this PR starting with build cf758f22 https://youtu.be/aknq4rr_dCM and then immediately broke again in the following build 687f4b55  https://youtu.be/avg8l5YO5HM.

**Comment by elad335** [comment]
That's because it was actually using interpreter due to a bug in that commit.

**Comment by chane2k1** [comment]
@elad335 That can't be completely true. I get a locked 60fps with LLVM enabled in that build and if I switch it back to fast interpreter https://youtu.be/6m6MWzlvPNA I get the expected ~40fps-60fps as in any other build.

**Comment by retrofan2019** [comment]
@chan2k1  yep i can confirm that it never runs at 60fps using the interpreter ... most of the time its sitting at 40fps  and it even can go down to 30fps ... only using LLVM it hits 60 fps stable but of course with the texture glitches .

File: 3rdparty/llvm.cmake [hunk 1]
```diff
@@ -53,7 +56,7 @@ if(WITH_LLVM)
 		endif()
 	endif()
 
-	set(LLVM_LIBS LLVMMCJIT LLVMX86CodeGen LLVMX86AsmParser)
+	set(LLVM_LIBS LLVMMCJIT LLVMX86CodeGen LLVMX86AsmParser LLVMIntelJITEvents)
```
Comment (jbeich) [inline 3rdparty/llvm.cmake:59]: Can you make this conditional like above? On FreeBSD with LLVM enabled (not on CI yet) I get
```
ld: error: unable to find library -lLLVMIntelJITEvents
```

**Comment by mew21dot** [comment]
why can kd 11 not put this into master build can plase merge this pull request

**Comment by RainbowCookie32** [comment]
The PR is not complete, and Neko himself can merge it whenever it is

**Comment by mew21dot** [comment]
his he ever going this and megere it is it ever going to happen

File: rpcs3_llvm.props [hunk 2]
```diff
@@ -10,7 +10,7 @@
...
       <AdditionalLibraryDirectories Condition="'$(Configuration)'=='Debug - LLVM'">%(AdditionalLibraryDirectories);..\llvm_build\Debug\lib</AdditionalLibraryDirectories>
       <AdditionalLibraryDirectories Condition="'$(Configuration)'=='Release - LLVM'">%(AdditionalLibraryDirectories);..\llvm_build\Release\lib</AdditionalLibraryDirectories>
-      <AdditionalDependencies>%(AdditionalDependencies);LLVMProfileData.lib;LLVMDebugInfoCodeView.lib;LLVMDebugInfoMSF.lib;LLVMInstrumentation.lib;LLVMMCJIT.lib;LLVMRuntimeDyld.lib;LLVMVectorize.lib;LLVMX86CodeGen.lib;LLVMGlobalISel.lib;LLVMX86Disassembler.lib;LLVMExecutionEngine.lib;LLVMAsmPrinter.lib;LLVMSelectionDAG.lib;LLVMCodeGen.lib;LLVMScalarOpts.lib;LLVMInstCombine.lib;LLVMTransformUtils.lib;LLVMAnalysis.lib;LLVMTarget.lib;LLVMX86Desc.lib;LLVMObject.lib;LLVMMCParser.lib;LLVMBitReader.lib;LLVMCore.lib;LLVMX86Utils.lib;LLVMMC.lib;LLVMX86Info.lib;LLVMSupport.lib;LLVMMCDisassembler.lib;LLVMipo.lib;LLVMBinaryFormat.lib;LLVMPasses.lib;LLVMIRReader.lib;LLVMLinker.lib;LLVMAsmParser.lib;LLVMX86AsmParser.lib;LLVMDemangle.lib;LLVMDebugInfoDWARF.lib;LLVMRemarks.lib;LLVMBitstreamReader.lib;LLVMTextAPI.lib;LLVMCFGuard.lib</AdditionalDependencies>
```
Comment (mew21dot) [inline rpcs3_llvm.props:13]: p

Comment (mew21dot) [inline rpcs3_llvm.props:13]: p

File: Utilities/JIT.h [hunk 3]
```diff
@@ -135,15 +136,12 @@ class jit_compiler final
 	// Local LLVM context
 	llvm::LLVMContext m_context;
 
-	// JIT Event Listener
-	std::unique_ptr<struct EventListener> m_jit_el;
```
Comment (mew21dot) [inline Utilities/JIT.h:139]: p

Comment (mew21dot) [reply Utilities/JIT.h:139]: can fix the problems and marigle into master build plaese

Comment (clienthax) [reply Utilities/JIT.h:139]: @mew21dot this is not a place for you to be a whiny arse, now stop spamming the tracker of things you don't understand.

**Comment by pppppppppyfv** [comment]
I am storry for spaming this pr with Mew21-dot it happed agign

**Comment by Mew99-dot** [comment]
Can anyone like someone works emulator fish this pr add master build have dayona usa  fixed

**Comment by Nekotekina** [comment]
Will be superseded by #9627

**Pull request closed without merging**